src/lib/stream/sink/types/zenoh_message.rs

```diff
@@ -0,0 +1,66 @@
+use serde::Serialize;
+
+/// A timestamp, represented as an offset from a user-defined epoch.
+/// <https://docs.foxglove.dev/docs/visualization/message-schemas/built-in-types#time>
+#[derive(Debug, Serialize)]
+pub struct Timestamp {
+    /// Seconds since epoch.
+    sec: u32,
+    /// Additional nanoseconds since epoch.
+    nsec: u32,
+}
+
+impl Timestamp {
+    pub fn new(sec: u32, nsec: u32) -> Self {
+        Self { sec, nsec }
+    }
+
+    pub fn now() -> Self {
+        Self {
+            sec: chrono::Utc::now().timestamp() as u32,
```

Comment:
```suggestion
            sec: chrono::Utc::now().timestamp() as i32,
```

Reply:
Yeah, I noticed that after commenting... I was using the schema definitions I had read, which were defining this field as an `i32` (as in, the `Time` type from [ROS](https://docs.ros2.org/foxy/api/builtin_interfaces/msg/Time.html)), in contrast with their [own implementation of Time](https://docs.foxglove.dev/docs/visualization/message-schemas/built-in-types#time). In their docs, they also mention Google's protobuf [Timestamp](https://protobuf.dev/reference/protobuf/google.protobuf/#timestamp), which defines it as an `i64` field.

Ultimately, the CDR should follow [their OMG IDL schema](https://github.com/foxglove/foxglove-sdk/blob/main/schemas/omgidl/foxglove/Time.idl): `u32`.